docs/aspnetcore-facility.md

```diff
@@ -8,40 +8,92 @@ The ASP.NET Core facility provides Castle Windsor integration using a custom act
+});
+```
+
+This is also useful if your framework components live in a separate assemblies or are not defined in the same assembly as your web application.
+Alternatively if your framework components all live in one assembly and you dont need to change lifestyles then you can simply use the following:
+
+```csharp
+services.AddWindsor(container, opts =>
+{
+	opts.UseEntryAssembly(typeof(HomeController).Assembly);
+});
+```
+This is good for trouble shooting situations where nothing get's registered because of problems in the hosting environment where 
+GetEntryAssembly/GetCallingAssembly does not work as expected. 
+
+### Registering custom middleware
+
+If you don't have any DI requirements in your middleware you can simply register it in the IServiceCollection like so: 
+
+```csharp
+services.AddSingleton<FrameworkMiddleware>(); // Do this if you don't care about using Windsor
+```
+
+Alternatively if you would like to inject dependencies into your middleware that are registered with Castle Windsor then you can use the 
+`UseMiddlewareFromWindsor` extension.
+
+```csharp
+// Add custom middleware, do this if your middleware uses DI from Windsor
+app.UseMiddlewareFromWindsor<CustomMiddleware>(container);
 ```
 
-It is also very important to note that you should never register any framework services in Windsor. This is handled by the framework for you. In the 
-case of [ILoggerFactory](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.iloggerfactory?view=aspnetcore-2.0) mentioned earlier, 
-you will notice it is not installed anywhere in the Startup.cs example below.
+### Cross Wiring into the IServiceCollection
 
-You also have an optional extension for validating that you don't have any types installed from the 'Microsoft.AspNetCore' called `AssertNoAspNetCoreRegistrations`.
-This could optionally be called from a unit test or you can add it to your Startup to guard against this at runtime.
+There is an additional feature you can use to `Cross Wire` components into the IServiceCollection. This is useful for cases where the framework
+needs to know how to resolve a type for example with the Razor @Inject directive in MVC projects. 
+
+```csharp
+container.Register(Component.For<IUserService>().ImplementedBy<AspNetUserService>().LifestyleScoped().CrossWired());
+```
+
+This would then allow you to consume the IUserService component in your Razor view like so:
+
+```html
+@inject WebApp.IUserService user
+
+@foreach (var user in user.GetAll())
+{
+	<div>
+		<h1>@user</h1>
+	</div>
+}
+```
+
+For this to work you have add the facility in the configure services method before you register anything. A helpful exception
+will be thrown here, in case you forget.
+
+```csharp
+public IServiceProvider ConfigureServices(IServiceCollection services)
+{
+	// Setup component model contributors for making windsor services available to IServiceProvider
+	Container.AddFacility<AspNetCoreFacility>(f => f.CrossWiresInto(services));
```

Comment:
@generik0 

Important breaking change. This is what eventually unlocked the key of being able to treat the WindsorContainer and the IServiceCollection as a tuple.